client/client-multi/client-c/src/kaa/platform-impl/common/encryption_utils.c

```diff
@@ -111,8 +111,9 @@ kaa_error_t ext_encrypt_data(const uint8_t *input, size_t payload_size, uint8_t
     size_t enc_data_size = ext_get_encrypted_data_size(payload_size);
     uint8_t padding = enc_data_size - payload_size;
     memset(output + payload_size, padding, padding);
+    memcpy(output, input, payload_size);
```

Comment:
Could you move this one line up? (That's more logical to start filling buffer from the start.)